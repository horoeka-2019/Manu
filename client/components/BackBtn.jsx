import React from 'react'
import { Button } from 'semantic-ui-react'

const BackBtn = () => <Button size='big' className='backBtnStyle'>Back</Button>

export default BackBtn
