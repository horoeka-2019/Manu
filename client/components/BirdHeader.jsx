import React from 'react'
import { Menu, Container } from 'semantic-ui-react'

class BirdHeader extends React.Component {
  render () {
    return (

      <Menu className='centerHeader' borderless fixed='top' inverted style = {{ backgroundColor: '#F3A712' }}>
        <Container>
          <Menu.Item>
            He Pokai Manu
          </Menu.Item>
        </Container>
      </Menu>

    )
  }
}

export default BirdHeader
