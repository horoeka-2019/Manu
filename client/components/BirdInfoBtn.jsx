import React from 'react'
import { Button } from 'semantic-ui-react'

const BirdInfoBtn = () => <Button size='huge' className='birdInfoBtnStyle'>Bird Info</Button>

export default BirdInfoBtn
