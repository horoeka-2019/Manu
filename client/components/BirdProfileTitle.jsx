import React from 'react'

class BirdProfileTitle extends React.Component {
  render () {
    return (
      <React.Fragment>

        <div className='birdProfileTitle'>Tui</div>

      </React.Fragment>
    )
  }
}

export default BirdProfileTitle
