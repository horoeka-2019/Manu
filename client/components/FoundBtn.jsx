import React from 'react'
import { Button } from 'semantic-ui-react'
import found from '../api/found'

const FoundBtn = () => <Button onClick= {() => found(this.props.id)} style = {{ marginBottom: '10px' }} size='massive' className='foundBtnStyle' >POKAI!</Button>

export default FoundBtn
