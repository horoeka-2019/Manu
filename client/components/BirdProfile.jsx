import React from 'react'
import BirdHeader from './BirdHeader'
import FoundBtn from './FoundBtn'
import BirdInfoBtn from './BirdInfoBtn'
import BackBtn from './BackBtn'
import BirdProfileTitle from './BirdProfileTitle'
import { Link } from 'react-router-dom'
import { Segment, Grid } from 'semantic-ui-react'
import found from '../api/found'

class BirdProfile extends React.Component {
  render () {
    const { id } = this.props.match.params
    const bird = this.props.birds.find(bird => bird.bird_id === Number(id))
    return (
      <React.Fragment>

        <Segment style={{ padding: '5px 5px' }} vertical>
          <Grid container stackable verticalAlign='middle'>
            <Grid.Row>
              <BirdHeader />
              <div className='birdProfileImage'>{bird.image}</div>
              <BirdProfileTitle name={bird.name}/>
              <Link to="/">
                <FoundBtn onClick={found(id)}/>
              </Link>
              <Link to={`/profile/${id}/info`}>
                <BirdInfoBtn />
              </Link>
              <Link to="/">
                <BackBtn />
              </Link>
            </Grid.Row>
          </Grid>

        </Segment>

      </React.Fragment>
    )
  }
}

export default BirdProfile
