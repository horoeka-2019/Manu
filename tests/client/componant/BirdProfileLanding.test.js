import React from 'react'
import { mount } from 'enzyme'
import BirdProfile from '../../../client/components/BirdProfile'
import BirdHeader from '../../../client/components/BirdHeader'
import BirdProfileTitle from '../../../client/components/BirdProfileTitle'
import BackBtn from '../../../client/components/BackBtn'
import FoundBtn from '../../../client/components/FoundBtn'
import BirdInfoBtn from '../../../client/components/BirdInfoBtn'

const mockBirdProfileProps = {
  image: 'public/images/piwakawaka.jpg/'
}

describe('BirdProfile component test', () => {
  it('BirdProfile contains header, image, found, bird info and back button components', () => {
    // Arrange
    const image = mockBirdProfileProps
    const expected = [<BirdHeader/>, image, <BirdProfileTitle />,
      <FoundBtn />,
      <BirdInfoBtn />,
      <BackBtn />]
    // Act
    const wrapper = mount(<BirdProfile/>)
    wrapper.setProps({ image: image })
    const actual = wrapper.containsAllMatchingElements(expected)
    // Assert
    expect(actual).toBeTruthy()
  })
 
})
