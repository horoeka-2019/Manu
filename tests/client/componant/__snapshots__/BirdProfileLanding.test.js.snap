// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BirdProfile component test render unchanged from previous snapshot 1`] = `
<BirdProfile>
  <Segment
    style={
      Object {
        "padding": "5px 5px",
      }
    }
    vertical={true}
  >
    <div
      className="ui vertical segment"
      style={
        Object {
          "padding": "5px 5px",
        }
      }
    >
      <Grid
        container={true}
        stackable={true}
        verticalAlign="center"
      >
        <div
          className="ui container stackable center aligned grid"
        >
          <GridRow>
            <div
              className="row"
            >
              <BirdHeader>
                <Menu
                  borderless={true}
                  className="centerHeader"
                  fixed="top"
                  inverted={true}
                  style={
                    Object {
                      "backgroundColor": "#F3A712",
                    }
                  }
                >
                  <div
                    className="ui borderless inverted top fixed centerHeader menu"
                    style={
                      Object {
                        "backgroundColor": "#F3A712",
                      }
                    }
                  >
                    <Container>
                      <div
                        className="ui container"
                      >
                        <MenuItem>
                          <div
                            className="item"
                            onClick={[Function]}
                          >
                            He Pokai Manu
                          </div>
                        </MenuItem>
                      </div>
                    </Container>
                  </div>
                </Menu>
              </BirdHeader>
              <div
                className="birdProfileImage"
              >
                <img
                  src="images/tui.png"
                />
              </div>
              <BirdProfileTitle>
                <div
                  className="birdProfileTitle"
                >
                  Tui
                </div>
              </BirdProfileTitle>
              <FoundBtn>
                <Button
                  as="button"
                  className="foundBtnStyle"
                  size="massive"
                  style={
                    Object {
                      "marginBottom": "10px",
                    }
                  }
                >
                  <Ref
                    innerRef={
                      Object {
                        "current": <button
                          class="ui massive button foundBtnStyle"
                          style="margin-bottom: 10px;"
                        >
                          POKAI!
                        </button>,
                      }
                    }
                  >
                    <RefFindNode
                      innerRef={
                        Object {
                          "current": <button
                            class="ui massive button foundBtnStyle"
                            style="margin-bottom: 10px;"
                          >
                            POKAI!
                          </button>,
                        }
                      }
                    >
                      <button
                        className="ui massive button foundBtnStyle"
                        onClick={[Function]}
                        style={
                          Object {
                            "marginBottom": "10px",
                          }
                        }
                      >
                        POKAI!
                      </button>
                    </RefFindNode>
                  </Ref>
                </Button>
              </FoundBtn>
              <BirdInfoBtn>
                <Button
                  as="button"
                  className="birdInfoBtnStyle"
                  size="huge"
                  style={
                    Object {
                      "marginBottom": "10px",
                    }
                  }
                >
                  <Ref
                    innerRef={
                      Object {
                        "current": <button
                          class="ui huge button birdInfoBtnStyle"
                          style="margin-bottom: 10px;"
                        >
                          Bird Info
                        </button>,
                      }
                    }
                  >
                    <RefFindNode
                      innerRef={
                        Object {
                          "current": <button
                            class="ui huge button birdInfoBtnStyle"
                            style="margin-bottom: 10px;"
                          >
                            Bird Info
                          </button>,
                        }
                      }
                    >
                      <button
                        className="ui huge button birdInfoBtnStyle"
                        onClick={[Function]}
                        style={
                          Object {
                            "marginBottom": "10px",
                          }
                        }
                      >
                        Bird Info
                      </button>
                    </RefFindNode>
                  </Ref>
                </Button>
              </BirdInfoBtn>
              <BackBtn>
                <Button
                  as="button"
                  className="backBtnStyle"
                  size="big"
                >
                  <Ref
                    innerRef={
                      Object {
                        "current": <button
                          class="ui big button backBtnStyle"
                        >
                          Back
                        </button>,
                      }
                    }
                  >
                    <RefFindNode
                      innerRef={
                        Object {
                          "current": <button
                            class="ui big button backBtnStyle"
                          >
                            Back
                          </button>,
                        }
                      }
                    >
                      <button
                        className="ui big button backBtnStyle"
                        onClick={[Function]}
                      >
                        Back
                      </button>
                    </RefFindNode>
                  </Ref>
                </Button>
              </BackBtn>
            </div>
          </GridRow>
        </div>
      </Grid>
    </div>
  </Segment>
</BirdProfile>
`;
